### PR TITLE
Fix certain solution asserts.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Review by ToddGrun — APPROVED** [review]
:shipit:

File: src/Workspaces/Core/Portable/Workspace/Solution/StateChecksums.cs [hunk 1]
```diff
@@ -486,6 +486,9 @@ public static ProjectStateChecksums Deserialize(ObjectReader reader)
             await ChecksumCollection.FindAsync(assetPath, state.AnalyzerConfigDocumentStates, searchingChecksumsLeft, onAssetFound, arg, cancellationToken).ConfigureAwait(false);
         }
     }
+
+    public override string ToString()
```
[the remark this answers is not in the document]
Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/Workspace/Solution/StateChecksums.cs:490]: yes.  i personally don't care one way or hte other.  it's all internal types, so ToString seems entirely reasonable to me.

[the remark this answers is not in the document]
Comment (ToddGrun) [reply src/Workspaces/Core/Portable/Workspace/Solution/StateChecksums.cs:490]: Yep, I'm fine either way. Whenever I see a ToString, I'm never certain if it's there for debugging or for actually having it's value being used in the code.

**Comment by CyrusNajmabadi** [comment]
@jasonmalinowski 	For review when you get back.

**Pull request merged**